src/oemof/solph/flows/_non_convex_flow_block.py

```diff
@@ -469,24 +475,22 @@ def min_downtime_rule(_, i, o, t):
             """
             Rule definition for min-downtime constraints of non-convex flows.
             """
-            if self._time_step_allows_flexibility(
-                t, m.flows[i, o].nonconvex.max_up_down, m.TIMESTEPS.at(-1)
-            ):
-                expr = 0
-                expr += (
-                    self.status[i, o, t - 1] - self.status[i, o, t]
-                ) * m.flows[i, o].nonconvex.minimum_downtime
-                expr += -m.flows[i, o].nonconvex.minimum_downtime
-                expr += sum(
-                    self.status[i, o, t + d]
-                    for d in range(0, m.flows[i, o].nonconvex.minimum_downtime)
+            expr = 0
+            expr += (
+                self.status[i, o, t] - self.status[i, o, t + 1]
```

Comment:
Hi @p-snft. I like your solution draft. I was wondering whether this line here does run into an indexing error if you don't limit it to t < m.TIMESTEPS.at(-1) and formulate the constraint over the TIMESTEPS set of the model?!

Reply:
Hey, I was thinking about the same thing. Maybe you can also reduce the number TIMESTEPS for the Constraint buliding

Reply:
I see that t+1 should not exist for `t == m.TIMESTEPS.at(-1)`, however for some reason it created the constraint. For backward compatibility, I'd rather test `if t > 0` so that we can continue working with
`self.status[i, o, t - 1] - self.status[i, o, t]`.